jira/client.go

```diff
@@ -0,0 +1,215 @@
+package jira
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+
+	"github.com/bitrise-io/go-utils/colorstring"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/urlutil"
+)
+
+const (
+	apiEndPoint     = "/rest/api/2/issue/"
+	commentEndPoint = "/comment"
+)
+
+// Client ...
+type Client struct {
+	token   string
+	client  *http.Client
+	headers map[string]string
+	url     string
+}
+
+// Comment ...
+type Comment struct {
+	Content string
+	IssuKey string
+}
+
+type response struct {
+	issueKey string
+	succes   bool
+	err      error
+}
+
+func (resp response) String() string {
+	respValue := map[bool]string{true: colorstring.Green("SUCCES"), false: colorstring.Red("FAILED")}[resp.succes]
+	return fmt.Sprintf("Posting comment to - %s - : %s", resp.issueKey, respValue)
+}
+
+// -------------------------------------
+// -- Public methods
+
+// NewClient ...
+func NewClient(token, requestURL string) *Client {
+	return &Client{token: token, client: &http.Client{}, headers: map[string]string{"Authorization": `Basic ` + token, "Content-Type": "application/json"}, url: requestURL}
+}
+
+// PostIssueComments ...
+func (client *Client) PostIssueComments(comments []Comment) error {
+	if len(comments) == 0 {
+		return fmt.Errorf("no comment has been added")
+	}
+
+	ch := make(chan response, len(comments))
+	for _, comment := range comments {
+		go client.postIssueComment(comment, ch)
+	}
+
+	counter := 0
+	var respErrors []response
+	for resp := range ch {
+		counter++
+		log.Printf(resp.String())
+
+		if resp.err != nil {
+			respErrors = append(respErrors, resp)
+		}
+
+		if counter >= len(comments) {
+			break
+		}
+	}
+
+	if len(respErrors) > 0 {
+		fmt.Println()
+		log.Infof("Errors during posting comments:")
+	}
+
+	for _, respErr := range respErrors {
+		log.Warnf("Error during posting comment to - %s - : %s", respErr.issueKey, respErr.err.Error())
+	}
+
+	if len(respErrors) > 0 {
+		fmt.Println()
+	}
+	return map[bool]error{true: fmt.Errorf("some comments were failed to be posted at Jira"), false: nil}[len(respErrors) > 0]
```

Comment:
no need to define the false key. if a key not defined, then the value's type's default value will be returned which is nil in this case.